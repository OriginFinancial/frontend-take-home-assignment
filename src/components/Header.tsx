import * as React from 'react';
import styled from 'styled-components';

import logo from '../icons/logo.svg';

const Container = styled.header`
  align-items: center;
  background: #fff;
  display: flex;
  height: 56px;
  padding: 0 16px;

  ${props => `
    ${props.theme.breakpoints.md} {
      height: 79px;
      padding: 0 37px;
    }
  `}
`;

const Logo = styled.img`
  height: 20px;

  ${props => `
    ${props.theme.breakpoints.md} {
      height: 30px;
    }
  `}
`;

const Header: React.FunctionComponent = () => {
  return (
    <Container>
      <Logo src={logo} alt="Origin Logo" />
    </Container>
  );
};

export default Header;
