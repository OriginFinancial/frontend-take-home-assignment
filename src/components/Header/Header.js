import React from 'react';
import styled from 'styled-components';
import { mq } from '../utils/mediaQueries';
import { node } from 'prop-types';

const propTypes = {
  children: node.isRequired
};

const HeaderStyled = styled.div`
  background-color: #fff;
  ${mq({
    textAlign: ['center', 'center', 'center', 'center', 'left']
  })}
  ${mq({
    padding: ['0', '0', '0', '0', '0 4em']
  })}
`;

const Header = ({ children }) => (
  <HeaderStyled data-testid="header-component">{children}</HeaderStyled>
);

Header.displayName = 'Header';
Header.propTypes = propTypes;

export { Header };
