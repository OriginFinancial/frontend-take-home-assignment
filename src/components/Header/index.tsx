import React from 'react';
import styled from 'styled-components';
import logo from '../../icons/logo.svg';

const HeaderStyle = styled.div`
  background: #fff;
  width: 100vw;
`;

const LogoStyle = styled.img`
  width: 65px;
  margin: 16px 29px 10px;

  @media (min-width: 1440px) {
    width: 95px;
  }
`;

const Header: React.FC = () => (
  <HeaderStyle>
    <LogoStyle src={logo} alt="Origin Logo" />
  </HeaderStyle>
);

export default Header;
