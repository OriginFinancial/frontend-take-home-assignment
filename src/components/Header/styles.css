header {
  width: 360px;
  height: 56px;
  left: 0px;
  top: 0px;
  background: #FFFFFF;
}