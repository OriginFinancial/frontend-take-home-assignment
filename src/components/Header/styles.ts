import styled from 'styled-components';

export const StyledHeader = styled.header`
  padding: 1.25rem 2.5rem;
  background-color: #fff;
`;
