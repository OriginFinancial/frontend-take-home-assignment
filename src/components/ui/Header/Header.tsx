import * as React from 'react';
import './header.scss';

const Header: React.FC = () => {
  return (
    <header className="mainHeader">
      <img src={require('~/assets/images/logoPrimary.svg')} alt="Origin" />
    </header>
  );
};

export default Header;
