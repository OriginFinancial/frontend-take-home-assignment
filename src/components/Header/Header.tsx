import * as React from 'react';
import styled from 'styled-components';
import { Container } from '..';
import { media } from '../../styles';
import OriginLogo from '../../icons/logo.svg';

const Header: React.FunctionComponent = () => {
  return (
    <Wrapper>
      <Container>
        <Logo src={OriginLogo} />
      </Container>
    </Wrapper>
  );
};

const Wrapper = styled.header`
  background-color: ${props => props.theme.colors.white};
  padding: 1.2rem 0.8rem;

  ${media.lg} {
    padding: 2rem 0;
  }
`;

const Logo = styled.img`
  color: ${props => props.theme.colors.text.main};
  width: 6rem;
  height: auto;
`;

export default Header;
