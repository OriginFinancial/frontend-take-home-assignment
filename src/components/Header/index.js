import React from 'react';

import './styles.css';

import logo from './src/icons/logo.svg';

function Header() {
  return (
    <header>
      <img src={logo} alt="Origin" />
    </header>
  );
}

export default Header;
