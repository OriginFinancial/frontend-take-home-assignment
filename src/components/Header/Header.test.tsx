import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import Header from './Header';

describe('Header component', () => {
  let component: ReactWrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    component = mount(<Header />);
  });

  it('renders initial layout', () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it('renders the logo image', () => {
    expect(component.find('img').length).toBe(1);
  });
});
