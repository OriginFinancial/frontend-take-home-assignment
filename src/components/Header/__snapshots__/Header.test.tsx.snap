// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders initial layout 1`] = `
Array [
  <Header />,
]
`;
